flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SourceTaskTerminationTest.java

```diff
@@ -118,9 +120,6 @@ private void stopWithSavepointStreamTaskTestHelper(final boolean shouldTerminate
 
             srcTaskTestHarness.processUntil(
                     srcTask.notifyCheckpointCompleteAsync(syncSavepointId)::isDone);
-            if (!shouldTerminate) {
-                assertFalse(srcTask.getSynchronousSavepointId().isPresent());
```

Comment:
Should we still keep this line to verify the syncSavepoint is cleaned up?

Reply:
The thing is it is not cleaned up, as it does not need to. The reason it was cleaned up was to break out of the synchronous mailbox loop. Without the loop there is no point in cleaning it up. We will exit anyway.